packages/calypso-e2e/src/media-helper.ts

```
@@ -45,61 +56,53 @@ export function getVideoDir(): string {
 /**
  * Creates a temporary test file by cloning a source file under a new name.
  *
+ * @param {string} sourcePath Full path on disk of the source file.
  * @param {{[key: string]: string}} param0 Parameter object.
- * @param {string} param0.sourceFileName Basename of the source file to be cloned.
- * @param {string} [param0.testFileName] Basename of the test file to be generated.
- * @returns {Promise<string>} Full path to the generated test file.
+ * @param {string} [param0.postfix] Additional suffix to be used for the file.
+ * @returns {Promise<TestFile>} Object implementing the TestFile interface.
+ * @throws {Error} If source file was not found, or source file did not contain an extension.
  */
-export async function createTestFile( {
-	sourceFileName,
-	testFileName,
-}: {
-	sourceFileName: string;
-	testFileName?: string;
-} ): Promise< string > {
-	let fileName = getTimestamp();
-	// If the output `testFileName` is defined, use that as part of the final filename.
-	if ( testFileName ) {
-		fileName += `-${ testFileName }`;
+export async function createTestFile(
+	sourcePath: string,
+	{
+		postfix,
+	}: {
+		postfix?: string;
+	} = {}
+): Promise< TestFile > {
+	try {
+		await fs.open( sourcePath, 'r' );
```

Could you explain what are we trying to do here? Seems weird to open a file descriptor and not use it (plus we should close the fd later). 

Is the goal to check that the file exists and is readable? In that case maybe we can use `fs.access()`

---

The intent is to check whether the file exists first. 
With that said, it seems something else is swallowing the error - an invalid path given to this method didn't output stack trace as I would have expected.

---

It looks like `throw Error` is being swallowed since the code calling `createTestFile` is within a hook. 

I've changed the `throw` statements to instead `console.error` and have verified this isn't swallowed up:

```
  console.error
    Source file /Users/edwintakahashi/workspace/wp-calypso-third/test/e2e/specs/image-uploads/image0.jpg not found on disk.

      80 | 		await fs.access( sourcePath );
      81 | 	} catch {
    > 82 | 		console.error( `Source file ${sourcePath} not found on disk.` );
         | 		        ^
      83 | 		// throw new Error( `Source file ${ sourcePath } not found on disk.` );
      84 | 	}
      85 |

      at Object.createTestFile (../../packages/calypso-e2e/src/media-helper.ts:82:11)
      at specs/specs-playwright/wp-blocks__media-spec.ts:27:11

```

This is a common topic of discussion as shown [here](https://github.com/facebook/jest/issues/8688).

---

Fixed in 5d4b3c3d4f724d40004abc238563b38f45e33daa.